sympy/geometry/line.py

```diff
@@ -270,19 +299,25 @@ def is_perpendicular(l1, l2):
         >>> l1, l2 = Line(p1, p2), Line(p1, p3)
         >>> l1.is_perpendicular(l2)
         True
-
         >>> p4 = Point(5, 3)
         >>> l3 = Line(p1, p4)
         >>> l1.is_perpendicular(l3)
         False
+        >>> from sympy import Point3D, Line3D
+        >>> p1, p2, p3 = Point3D(0, 0, 0), Point3D(1, 1, 1), Point3D(-1, 2, 0)
+        >>> l1, l2 = Line3D(p1, p2), Line3D(p2, p3)
+        >>> l1.is_perpendicular(l2)
+        False
+        >>> p4 = Point3D(5, 3, 7)
+        >>> l3 = Line3D(p1, p4)
+        >>> l1.is_perpendicular(l3)
+        False
 
         """
-        try:
-            a1, b1, c1 = l1.coefficients
-            a2, b2, c2 = l2.coefficients
-            return bool(simplify(a1*a2 + b1*b2) == 0)
-        except AttributeError:
-            return False
+        if not isinstance(l1, LinearEntity) and not isinstance(l2, LinearEntity):
+            raise TypeError('Must pass only LinearEntity objects')
+
+        return l1.direction.dot(l2.direction) == S.Zero
```

Comment:
Isn't this made weaker by not including a call to `simplify` -- `dot` doesn't do simplification? I wonder if returning an `Eq` which might be unevaluated might be better. One would have to see how this is handled elsewhere.